### PR TITLE
Adding several products

### PR DESCRIPTION
**Pull request by Fraseer**
PT Link: https://www.pivotaltracker.com/story/show/179716933

### User Story:
As a user
In order to order the food that I want
I would like to be able update my cart with more products

### Changes Proposed
- Create tests for updating the cart with a new product
- Implement a update method in the cart controller
- Update message response to include new item and http status

### What we learned working on this feature
Keep it in scope
Cart.find(params['id']) & current_user.carts dont work the same way
Setting the right premises in the before blocks for testing

### Packages / Gems added
N/A

### Screenshots (Videos)
image

### CODE REVIEW

File: spec/requests/api/api_adds_several_products_to_the_cart_spec.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+          params: {
+            product_id: product.id,
+          },
+          headers: auth_headers
+    end
+
+    it { is_expected.to have_http_status 200 }
+
+    it 'is expected to return an array of items' do
+      # binding.pry
```
Comment (lukeperrin10) [inline spec/requests/api/api_adds_several_products_to_the_cart_spec.rb]: This should be removed.

**Review by lukeperrin10 — CHANGES_REQUESTED** [review]
Please remove the binding.pry otherwise looks good

File: spec/requests/api/api_adds_several_products_to_the_cart_spec.rb (hunk 1 above)
Comment (Fraseer) [reply spec/requests/api/api_adds_several_products_to_the_cart_spec.rb]: I have pushed up the change

**Review by lukeperrin10 — APPROVED** [review]

File: spec/requests/api/api_adds_several_products_to_the_cart_spec.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
+RSpec.describe 'PUT /api/carts', type: :request do
+  subject { response }
+  let(:product_existing) { create(:product, name: 'Kangaroo Steak') }
+  let(:product) { create(:product, name: 'Whale Blubber Stew') }
+  let(:user) { create(:user) }
+  let(:cart) { create(:cart, user_id: user.id) }
```
Comment (tochman) [inline spec/requests/api/api_adds_several_products_to_the_cart_spec.rb:6]:
```
let(:cart) { create(:cart, user: user) }
```

File: spec/requests/api/api_adds_several_products_to_the_cart_spec.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  let(:auth_headers) { user.create_new_auth_token }
+  describe 'authorized user can add products to an existing cart' do
+    before do
+      cart.cart_products.create(product_id: product_existing.id)
```
Comment (tochman) [inline spec/requests/api/api_adds_several_products_to_the_cart_spec.rb:10]:
```
cart.cart_products.create(product: product_existing)
```

File: spec/requests/api/api_adds_several_products_to_the_cart_spec.rb [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      put "/api/carts/#{cart.id}",
```
Comment (tochman) [inline spec/requests/api/api_adds_several_products_to_the_cart_spec.rb:11]:
```
put carts_path(cart)
```

**Pull request merged**